### ISSUE TITLE
Using Django Password Validators

### ISSUE THREAD

**Issue by pik** [sunscrapers/djoser#147]
I see Djoser has it's own method of providing password validation which does not use the Django `AUTH_PASSWORD_VALIDATORS` and will not accept identical arguments. Is there a reason for this? Wouldn't it be simpler to work directly with what is provided by Django auth (https://docs.djangoproject.com/en/1.9/topics/auth/passwords/) ?

**Comment by mechtron** [comment]
+1

**Comment by rawberg** [comment]
Looks like #107 was about to add this functionality but got combined with some other PRs and didn't end up getting merged unfortunately. I was able to follow the code in that commit and get it working pretty quickly by extending UserRegistrationSerializer and just overriding the password field as shown in the commit for #107.

**Comment by aspcanada** [comment]
How do I set the validators that the rest of my app uses in AUTH_PASSWORD_VALIDATORS?

**Comment by msolima** [comment]
You can do this in your settings.py

```
AUTH_PASSWORD_VALIDATORS = [
    {
        'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.CommonPasswordValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
    },
]
```

Then also in settings.py for DJOSER Settings:

```

from django.contrib.auth import password_validation

DJOSER = {
    'PASSWORD_RESET_CONFIRM_URL': '#/password/reset/confirm/{uid}/{token}',
    'ACTIVATION_URL': '#/activate/{uid}/{token}',
    'PASSWORD_VALIDATORS': [password_validation.validate_password]
}
```

works for me!

**Comment by rvignesh89** [comment]
@msolima your fix worked for me as well. However when I wanted to add `UserAttributeSimilarityValidator` into the mix I realised that the fix was not that simple. It is because `UserAttributeSimilariryValidator` requires the User instance to be passed in. This does not happen currently since djoser uses serializer field validation which does not have access to user instance.

I like @pik's idea of re-using the `AUTH_PASSWORD_VALIDATORS ` settings in djoser considering DRY principles. Some digging into Django [documentation](https://docs.djangoproject.com/en/1.11/topics/auth/passwords/#integrating-validation) suggests us to use `validate_password` if we are writing our own mechanisms to change password which is exactly whats being done here.

Considering this I think we can implement password validation in djoser something like below and remove populating password validators in djoser settings again.


```
class PasswordResetConfirmView(utils.ActionViewMixin, generics.GenericAPIView):
   ...
    def _action(self, serializer):
        validate_password(serializers.data['new_password'], serializer.user)
        serializer.user.set_password(serializer.data['new_password'])
        serializer.user.save()
        return Response(status=status.HTTP_204_NO_CONTENT)
```

@piotr-szpetkowski what do you think?

**Comment by piotr-szpetkowski** [comment]
Awesome ideas everyone!
I also strongly believe it would be best to use Django's own password validation, however there is a problem that it has been added in [1.9 release](https://docs.djangoproject.com/en/1.11/releases/1.9/#password-validation), while 1.8 will still be supported at least until April 2018 since it's LTS version.

There are few solutions for this problem that I see at the moment:
1. Stop supporting Django 1.8 - quite dramatic, yet very effective.
2. Keep existing `PASSWORD_VALIDATORS` in Djoser as a fallback for 1.8 and use `django.contrib.auth.password_validation` otherwise.

I personally believe that going for second option would be best. In that way we might deprecate `PASSWORD_VALIDATORS` somewhere along the road when Django 1.8 stops being supported.

**Comment by piotr-szpetkowski** [comment]
After discussing the matter internally with @haxoza and @michalbon we believe that https://github.com/orcasgit/django-password-validation will be a solution for for Django 1.8 users, so I suppose that we might as well get rid of `PASSWORD_VALIDATORS` from Djoser settings for 0.7.0 and we will simply use Django Password Validation as we should.

**Comment by piotr-szpetkowski** [comment]
Fixed with #205
